plugins/security/nexus-ldap-plugin/src/test/java/org/sonatype/security/realms/ldap/internal/LdapTestSupport.java

```diff
@@ -13,136 +13,280 @@
-
-    ldapServer = (LdapServer) lookup(LdapServer.ROLE);
-    ldapRealmConfig = new File(getConfHomeDir(), "ldap.xml");
-    if (sourceLdapXml.exists()) {
-      this.interpolateLdapXml(sourceLdapXml, ldapRealmConfig);
+    ldapServerConfigurations = createLdapServerConfigurations();
+    for (Map.Entry<String, LdapServerConfiguration> configurationEntry : ldapServerConfigurations.entrySet()) {
+      final LdapServer ldapServer = new LdapServer(configurationEntry.getValue());
+      ldapServers.put(configurationEntry.getKey(), ldapServer);
     }
-    else {
-      this.interpolateLdapXml("/test-conf/etc/ldap.xml", ldapRealmConfig);
+    startLdapServers();
+    ldapClientConfigurations = createLdapClientConfigurations();
+    for (LdapConfiguration configuration : ldapClientConfigurations.values()) {
+      ldapConfigurationSource.create(configuration);
     }
   }
 
-  @Override
-  public void tearDown()
+  protected synchronized void startLdapServers() {
+    for (String name : ldapServerConfigurations.keySet()) {
+      startLdapServer(name);
+    }
+  }
+
+  protected synchronized void startLdapServer(final String name) {
+    checkArgument(ldapServers.containsKey(name), "Unknown LDAP Server %s", name);
+    final LdapServer ldapServer = ldapServers.get(name).start();
+  }
+
+  @After
+  public void stopLdap()
       throws Exception
   {
     lookup(CacheManager.class).shutdown();
+    stopLdapServers();
+  }
 
-    if (ldapServer != null && ldapServer.isStarted()) {
-      ldapServer.stop();
-      ldapServer = null;
+  protected synchronized void stopLdapServers() {
+    for (String name : ldapServerConfigurations.keySet()) {
+      stopLdapServer(name);
     }
-
-    super.tearDown();
   }
 
-  @Override
-  protected void customizeContainerConfiguration(final ContainerConfiguration configuration) {
-    super.customizeContainerConfiguration(configuration);
-    configuration.setAutoWiring(true);
-    configuration.setClassPathScanning(PlexusConstants.SCANNING_INDEX);
+  protected synchronized void stopLdapServer(final String name) {
+    checkArgument(ldapServerConfigurations.containsKey(name), "Unknown LDAP Server %s", name);
+    final LdapServer ldapServer = ldapServers.get(name);
+    if (ldapServer != null) {
+      ldapServer.stop();
+    }
   }
 
-  /**
-   * Copies a resource from from the classpath to the given output file and closes streams.
-   */
-  protected void copyResourceToFile(String resource, File outputFile)
-      throws IOException
-  {
-    try (InputStream in = getClass().getResourceAsStream(resource);
-         OutputStream out = new FileOutputStream(outputFile)) {
-      IOUtils.copy(in, out);
+  protected LinkedHashMap<String, LdapConfiguration> createLdapClientConfigurations() {
+    final LinkedHashMap<String, LdapConfiguration> result = Maps.newLinkedHashMap();
+    // default implementation creates entries for all defined LDAP servers
+    int order = 0;
+    for (String name : ldapServers.keySet()) {
```

Comment:
entrySet() would be preferable given the first thing you do is look up the value